portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -70,19 +74,26 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String name = getParameter(request, "name-input", "");
     String text = getParameter(request, "text-input", "");
     boolean upperCase = Boolean.parseBoolean(getParameter(request, "upper-case", "false"));
+    long timestamp = System.currentTimeMillis();
 
     // Convert the text to upper case.
     if (upperCase) {
       text = text.toUpperCase();
     }
 
-    long timestamp = System.currentTimeMillis();
+    // Sentiment Analysis of the comment
+    Document doc = Document.newBuilder().setContent(text).setType(Document.Type.PLAIN_TEXT).build();
```

Comment:
Just out of curiosity, have you seen this kind of pattern before, where an object is created by Class.newBuilder().....build()? If not then I can ramble about it a bit, because it's a useful pattern. (It's called, perhaps unsurprisingly, the Builder pattern.)

Reply:
I have not seen this pattern before. I'd like an explanation please